### ISSUE TITLE
how to calculate the mean and std of the image of the  datasets which is my own datasets?

### ISSUE THREAD

**Issue by liufengwei520** [zylo117/Yet-Another-EfficientDet-Pytorch#409]
I'm new  in this field ,please help me . please ,please

**Comment by picEmily** [comment]
`import torch
import torchvision
from torch.utils.data import DataLoader
from torchvision import transforms, datasets
import matplotlib.pyplot as plt
import numpy as np

data_transform = transforms.Compose([
    transforms.ToTensor(),   
])

testdataset = datasets.ImageFolder(
    root='./data/mydata', transform=data_transform)

dataloader = torch.utils.data.DataLoader(
    testdataset, batch_size=1, shuffle=True, num_workers=2)
mean = torch.zeros(3)
std = torch.zeros(3)
print('==> Computing mean and std..')
for inputs, targets in dataloader:
    for i in range(3):
        mean[i] += inputs[:, i, :, :].mean()
        std[i] += inputs[:, i, :, :].std()
mean.div_(len(testdataset))
std.div_(len(testdataset))
print(mean)
print(std)`

你看这样行不？

**Comment by picEmily** [comment]
```python
import torch
import torchvision
from torch.utils.data import DataLoader
from torchvision import transforms, datasets
import matplotlib.pyplot as plt
import numpy as np

data_transform = transforms.Compose([
    transforms.ToTensor(),   
])

testdataset = datasets.ImageFolder(
    root='./data/mydata', transform=data_transform)

dataloader = torch.utils.data.DataLoader(
    testdataset, batch_size=1, shuffle=True, num_workers=2)
mean = torch.zeros(3)
std = torch.zeros(3)
print('==> Computing mean and std..')
for inputs, targets in dataloader:
    for i in range(3):
        mean[i] += inputs[:, i, :, :].mean()
        std[i] += inputs[:, i, :, :].std()
mean.div_(len(testdataset))
std.div_(len(testdataset))
print(mean)
print(std)
```
这样可以计算mean和std

**Comment by liufengwei520** [comment]
Thank for your assitance .you're a great and good person.Hnnn
